pkg/services/ngalert/state/persister_async.go

```diff
@@ -15,20 +15,22 @@ type AsyncStatePersister struct {
 	// doNotSaveNormalState controls whether eval.Normal state is persisted to the database and returned by get methods.
 	doNotSaveNormalState bool
 	store                InstanceStore
+	ticker               *clock.Ticker
 }
 
-func NewAsyncStatePersister(log log.Logger, cfg ManagerCfg) StatePersister {
+func NewAsyncStatePersister(log log.Logger, ticker *clock.Ticker, cfg ManagerCfg) StatePersister {
 	return &AsyncStatePersister{
 		log:                  log,
 		store:                cfg.InstanceStore,
+		ticker:               ticker,
 		doNotSaveNormalState: cfg.DoNotSaveNormalState,
 	}
 }
 
-func (a *AsyncStatePersister) Async(ctx context.Context, ticker *clock.Ticker, cache *cache) {
+func (a *AsyncStatePersister) Async(ctx context.Context, cache *cache) {
```

Comment:
Something that I think is missing here is there are no metrics to tell me how long a full sync takes, so can't know if `state_periodic_save_interval` is too low and needs to be increased (or if it's too high and can be reduced).

Reply:
I don't think it even needs to be a metric thinking about it again, can just be a log line?

Reply:
Fair enough! Will add a metric.

Reply:
```
# HELP grafana_alerting_state_full_sync_duration_seconds The duration of fully synchronizing the state with the database.
# TYPE grafana_alerting_state_full_sync_duration_seconds histogram
grafana_alerting_state_full_sync_duration_seconds_bucket{le="0.01"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="0.1"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="1"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="2"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="5"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="10"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="60"} 2
grafana_alerting_state_full_sync_duration_seconds_bucket{le="+Inf"} 2
grafana_alerting_state_full_sync_duration_seconds_sum 0.006880625
grafana_alerting_state_full_sync_duration_seconds_count 2
```